api/handlers.go

```diff
@@ -15,6 +16,7 @@ import (
 	log "github.com/sirupsen/logrus"
 
 	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions/logic"
```

Comment:
We need to avoid adding new dependencies to go-algorand

Reply:
I wanted to enforce the same box encoding behavior as we use in goal and algod, so it seemed prudent to use  the pre-existing `logic.NewAppCallBytes(encodedBoxName)`. I could remove the dependency by re-creating the function in indexer.

Reply:
We're going to remove the submodule at some point, which is why I bring it up.

Reply:
@winder I'll ultimately follow your guidance here, but want to voice my preference to keep as is.  Does the following rationale work for you?

Rationale:
* I understand we'll move away from the submodule, but the work in-question began prior to the course change.   Since there's multiple changes needed to remove the submodule dependency, the incremental addition here feels marginal.
* Keeping as is makes it easier to understand sources of potential duplication.  If the dependency is dropped in favor of duplication, I think it will be difficult to make a pass later on identifying the duplication.

Reply:
I see 3 functions: `logic.NewAppCallBytes`, `logic.SplitBoxKey` and `logic.MakeBoxKey`. They all appear to be small utility functions that would be straightforward to copy.

* The first point doesn't seem fair... it kind of boils down to "you've already got a lot of work ahead, whats a little more?"
* The second makes more sense. Are you imagining that these functions might be pulled out in a similar way to the abi module?

Reply:
@winder Let's focus on the 2nd rationale:  Yes - It feels plausible to me that we'll wish to extract common functionality to license appropriately + minimize the chance of divergence.

Reply:
Ok, that makes sense. Let's go with your approach and punt on this for now.